src/Native/System.Private.CoreLib.Native/pal_environment.cpp

```diff
@@ -3,28 +3,43 @@
 // See the LICENSE file in the project root for more information.
 #include <stdlib.h>
 #include <string.h>
+#include <unistd.h>
 #include <assert.h>
 
 extern "C" int32_t CoreLibNative_GetEnvironmentVariable(const char* variable, char** result)
 {
-   assert(result != NULL);
+    assert(result != NULL);
 
-   // Read the environment variable
-   *result = getenv(variable);
+    // Read the environment variable
+    *result = getenv(variable);
 
-   if (*result == NULL)
-   {
-       return 0;
-   }
+    if (*result == NULL)
+    {
+        return 0;
+    }
 
-   size_t resultSize = strlen(*result);
+    size_t resultSize = strlen(*result);
 
-   // Return -1 if the size overflows an integer so that we can throw on managed side
-   if ((size_t)(int32_t)resultSize != resultSize)
-   {
-       *result = NULL;
-       return -1;
-   }
+    // Return -1 if the size overflows an integer so that we can throw on managed side
+    if ((size_t)(int32_t)resultSize != resultSize)
+    {
+        *result = NULL;
+        return -1;
+    }
 
-   return (int32_t)resultSize;
+    return (int32_t)resultSize;
+}
+
+extern "C" int32_t CoreLibNative_GetHostName(char* hostNameBuffer, int32_t hostNameBufferLength)
+{
+    assert(hostNameBuffer != NULL && hostNameBufferLength > 0);
+
+    int32_t res = gethostname(hostNameBuffer, hostNameBufferLength);
+    if (res < 0)
+        return res;
+
+    // gethostname should store the null terminator but we set the last slot in the buffer to null
+    // to be protected against buffer overflow when asking for the length
+    hostNameBuffer[hostNameBufferLength - 1] = '\0';
```

Comment:
BTW: System.Native in corefx has wrapper for gethostname as well, but it does not have protection like this.